### PR TITLE
Homogenise presentation format to indy (Dict) in PresentationExchange

### LINKED ISSUE #183: unify presentation value type for PresentationExchange v1/v2

### PR DESCRIPTION
**Pull request by morrieinmaas**
Add presentation_request to PresentationExchange

Signed-off-by: morrieinmaas <moritz@animo.id>

closes #183

### CODE REVIEW

File: app/generic/verifier/facades/acapy_verifier_utils.py [hunk 1]
```diff
@@ -23,7 +23,10 @@ def record_to_model(
             connection_id=record.connection_id,
             created_at=record.created_at,
             protocol_version=ProofRequestProtocolVersion.v20.value,
-            presentation=record.pres,
+            presentation=record.by_format.pres["indy"]
+            if record.by_format.pres
+            else record.by_format.pres,
```
Comment (morrieinmaas) [inline app/generic/verifier/facades/acapy_verifier_utils.py:28]: If this is None (no presentation was provided)

File: app/generic/verifier/facades/acapy_verifier_utils.py [hunk 2]
```diff
@@ -23,7 +23,10 @@ def record_to_model(
...
+            presentation_request=record.by_format.pres_request["indy"],
```
Comment (morrieinmaas) [inline app/generic/verifier/facades/acapy_verifier_utils.py:29]: Request was still made

File: app/generic/verifier/facades/acapy_verifier_utils.py (hunk 1 above)
Comment (TimoGlastra) [inline app/generic/verifier/facades/acapy_verifier_utils.py:28]: `record.by_format.pres` will always be `None` right? In that case we can maybe use `None` to make it more explicit?

```suggestion
            else None
```

File: app/generic/verifier/models.py [hunk 3]
```diff
@@ -45,9 +42,8 @@ class PresentationExchange(BaseModel):
     connection_id: Optional[str] = None
     created_at: str
     proof_id: str
-    presentation: Optional[
-        Union[IndyProof, V20Pres, V20PresExRecordByFormat, Dict]
-    ] = None
+    presentation: Optional[Dict] = None
+    presentation_request: Optional[Dict] = None
```
Comment (TimoGlastra) [inline app/generic/verifier/models.py:46]: If we use the model types here we would also have typing in the swagger ui:

```suggestion
    presentation: Optional[IndyProof] = None
    presentation_request: Optional[IndyProofRequest] = None
```

**Review by TimoGlastra — COMMENTED** [review]
Nice! two nitpicks

File: app/generic/verifier/models.py (hunk 3 above)
Comment (morrieinmaas) [reply app/generic/verifier/models.py:46]: sure I'll give that a try. I think aca-py returns a dict so I'll have to construct that from the dict. will give it a go

File: app/generic/verifier/facades/acapy_verifier_utils.py (hunk 1 above)
Comment (morrieinmaas) [reply app/generic/verifier/facades/acapy_verifier_utils.py:28]: fair point

File: app/generic/verifier/models.py (hunk 3 above)
Comment (TimoGlastra) [reply app/generic/verifier/models.py:46]: We do have to transform the dict to a model then in the v2 record_to_model  transformation

**Pull request merged**